apollo-api/src/commonMain/kotlin/com/apollographql/apollo3/api/AdapterContext.kt

```diff
@@ -19,7 +19,11 @@ class AdapterContext private constructor(
   }
 
   fun hasDeferredFragment(path: List<Any>, label: String?): Boolean {
-    return mergedDeferredFragmentIds?.contains(DeferredFragmentIdentifier(path, label)) == true
+    if (mergedDeferredFragmentIds == null) {
+      // By default, parse all deferred fragments - this is the case when parsing from the normalized cache.
+      return true
+    }
+    return mergedDeferredFragmentIds.contains(DeferredFragmentIdentifier(path, label))
```

Comment:
This means that reading from the cache will basically ignore the `@defer`, right? It'd be interesting to call this out somewhere, future documentation I guess

Reply:
Yes exactly. If some fields are not yet received, this will be a cache miss, otherwise, we read all the fragments as if `@defer` didn't exist.